### PR TITLE
Adapt service for the new development environment (no 3rd party tools required)

### PR DESCRIPTION
**Pull request by dgaubert**
In order to run tests in the new development environment:
- stop using pg tools (`psql`, `pg_dump`, ...) and redis-cli to setup fixtures
- clean tests leftovers
- adapt SQL scripts to be run with `node-postgres`
- adapt Redis set up to be run with `node-redis` 
- fix tests where they create keys in Redis that weren't cleaned

Note: I guess we can stop using dedicated images for testing purposes

### CODE REVIEW

File: test/support/test-helper.js [hunk 1]
```diff
@@ -69,77 +69,48 @@ function checkSurrogateKey (res, expectedKey) {
...
+    for (const db of Object.keys(expectedKeys)) {
+        await new Promise((resolve, reject) => client.select(db, (err) => err ? reject(err) : resolve()));
+        const keys = await new Promise((resolve, reject) => client.keys('*', (err, keys) => err ? reject(err) : resolve(keys)));
+        foundKeys[db].push(...keys);
+    }
 
-        if (Object.keys(databasesTasks).length === 0) {
-            assert.strictEqual(keysFound.length, 0, 'Unexpected keys found in redis: ' + keysFound.join(', '));
-            done();
+    const unexpectedKeys = { 0: [], 5: [] };
```
Comment (oleurud) [inline test/support/test-helper.js:98]: should we add here 8 and 3?

Comment (oleurud) [reply test/support/test-helper.js:98]: Answering myself, I think the rate limit keys (db 8) are well removed in tests, but maybe is a good thing, in any case, to check them here

**Review by oleurud — APPROVED** [review]
I see it fine. I added 2 "just in case questions". 

I have not checked the SQL files, since tests are working I guess fixtures are fine (and that the change has not been too painful for you)

File: test/index.js [hunk 2]
```diff
@@ -56,87 +71,131 @@ async function populateDatabase () {
...
+        '_vovw_1_test_special_float_values_table_overviews',
+        'test_table_localhost_regular1',
+        'analysis_banks',
+        'analysis_rent_listings',
+        'test_table_100',
+        'test_table_200k',
+        'populated_places_simple_reduced',
+        'populated_places_simple_reduced_private'
+    ];
+    await query({ db: TEST_DB, sql: `VACUUM ANALYZE ${tables.join(', ')}` });
```
[the remark this answers is not in the document]
Comment (dgaubert) [reply test/index.js:109]: To stabilize tests, it's legacy. Not sure if we still need it.

File: test/support/test-helper.js (hunk 1 above)
Comment (dgaubert) [reply test/support/test-helper.js:98]: Yeah, I'm going to check them, the main goal is to leave Redis as it was before running the tests.

File: test/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (oleurud) [reply test/index.js:109]: Ahh, I didn't remember that I did, so I didn't say anything.

File: test/support/test-helper.js (hunk 1 above)
Comment (dgaubert) [reply test/support/test-helper.js:98]: the DB 3, do you know for what it is being used?

Comment (oleurud) [reply test/support/test-helper.js:98]: Old "oauth" stuff https://github.com/CartoDB/node-cartodb-redis/blob/master/lib/carto_metadata.js#L48

Probably it is not in use, at least I know nothing about it.

Comment (oleurud) [reply test/support/test-helper.js:98]: It seems to be used in SQL API https://github.com/CartoDB/CartoDB-SQL-API/blob/master/lib/auth/oauth.js, but not here

**Review by ilbambino — APPROVED** [review]
I like the changes, 👍 💯

**Pull request merged**